### PR TITLE
Enable is_stable argument in top_k Ops

### PR DESCRIPTION
**Pull request by gaga1313**
## Description
This PR integrates the `is_stable` optional boolean argument (defaulting to `True`) into `keras.ops.top_k` and `keras.ops.TopK` across all five supported backends (`tensorflow`, `jax`, `torch`, `numpy`, and `openvino`). When `is_stable=True`, equal elements in the input tensor preserve their original left-to-right relative order in the returned indices.

## Backend Implementations
- **TensorFlow**: Uses native `tf.math.top_k(x, k, sorted=sorted)` (inherently stable on CPU).
- **JAX**: Forwards `is_stable=is_stable` to `jax.lax.top_k`, with a defensive `try...except TypeError` fallback for older JAX releases that lack the keyword argument.
- **PyTorch**: Uses `torch.sort(x, dim=-1, descending=True, stable=True)[:k]` when `is_stable=True`, and fast `torch.topk(x, k, sorted=sorted)` when `is_stable=False`. Also updates `in_top_k` tuple indexing (`[0]`) for return type compatibility.
- **NumPy**: Implements universal Reverse-Argsort-Reverse stable sorting (`np.flip` + `np.argsort(kind="stable")[..., ::-1]`) when `is_stable=True`.
- **OpenVINO**: Forwards `stable=is_stable` to `ov_opset.topk`, setting `sort_type="value"` whenever `is_stable=True` to satisfy OpenVINO node validation requirements.

## Backward Compatibility & Serialization
- Existing positional and keyword calls (`top_k(x, k)`, `top_k(x, k, sorted=False)`) function without modification.
- `TopK.get_config()` serializes `'is_stable'`, and deserializing legacy configs gracefully defaults to `is_stable=True`.

## Testing
- Added a vectorized `check_stability(self, values, indices)` verification helper in `src/ops/math_test.py` that checks ascending index order across duplicate entries along the last axis.
- Added comprehensive unit tests:
  - `test_top_k_stability_small`: Small 1D array verifying exact index preservation (`[3, 5, 5, 2, 5, 1]` -> `[1, 2, 4, 0]`).
  - `test_top_k_stability_large_1d`: Stress test on 10,000-element 1D arrays with many duplicates.
  - `test_top_k_stability_large_2d`: Stress test on multi-dimensional `(10, 500)` arrays with duplicates.
  - `test_top_k_unstable_large`: Verifies `is_stable=False`.
  - `test_top_k_operation_config_and_symbolic_call`: Verifies `TopK(Operation)` symbolic calls and `get_config()` serialization.
All Keras 3 backends (`jax`, `tensorflow`, `torch`, `numpy`, `openvino`) now correctly support `sorted=False`.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/backend/torch/math.py [hunk 1]
```diff
@@ -77,16 +77,21 @@ def segment_prod(data, segment_ids, num_segments=None, sorted=False):
...
+    return sorted_values[..., :k], sorted_indices[..., :k]
 
 
 def in_top_k(targets, predictions, k):
     targets = convert_to_tensor(targets).type(torch.int64)
     targets = targets.unsqueeze(-1)
     predictions = convert_to_tensor(predictions)
-    topk_values = top_k(predictions, k).values
+    topk_values = top_k(predictions, k)[0]
     targets_values = torch.take_along_dim(predictions, targets, dim=-1)
```
Comment (hertschuh) [inline keras/src/backend/torch/math.py]: Pass `is_stable=False` here. Stability is not needed since the indices returned are completely ignored.

In fact, you could pass `sorted=False` too because it's comparing the `target_values` to **all** the topk values and doing `any` (and not just comparing with the kth value).

Please do the same in the NumPy backend.

File: keras/src/backend/jax/math.py [hunk 2]
```diff
@@ -52,10 +52,13 @@ def segment_prod(data, segment_ids, num_segments=None, sorted=False):
...
 
 
-def top_k(x, k, sorted=True):
-    # Jax does not supported `sorted`, but in the case where `sorted=False`,
+def top_k(x, k, sorted=True, is_stable=True):
+    # Jax does not support `sorted`, but in the case where `sorted=False`,
     # order is not guaranteed, so OK to return sorted output.
-    return jax.lax.top_k(x, k)
+    try:
+        return jax.lax.top_k(x, k, is_stable=is_stable)
```
Comment (hertschuh) [inline keras/src/backend/jax/math.py:60]: Please add a comment saying that `is_stable` was added in JAX 0.11.0

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for adding this!

File: keras/src/backend/torch/math.py (hunk 1 above)
Comment (gaga1313) [reply keras/src/backend/torch/math.py]: Thanks for catching this!

File: keras/src/backend/jax/math.py (hunk 2 above)
Comment (gaga1313) [reply keras/src/backend/jax/math.py:60]: fixed!

**Review by gaga1313 — COMMENTED** [review]
Thanks for the review!

**Review by hertschuh — APPROVED** [review]
Thanks for adding this!
